RealmSwift/BSON.swift

```diff
@@ -62,15 +62,23 @@ extension UUID: BSON {
 /// A Dictionary object representing a `BSON` document.
 public typealias Document = Dictionary<String, AnyBSON?>
 
-extension Dictionary: BSON where Key == String, Value == AnyBSON? {
+extension Dictionary: BSON, PartitionValue where Key == String, Value == AnyBSON? {
 }
 
-extension Array: BSON where Element == AnyBSON? {
+extension Array: BSON, PartitionValue where Element == AnyBSON? {
 }
 
 extension NSRegularExpression: BSON {
 }
 
+/// Implementation for `PartitionValue` conformance.
+extension BSON {
+    /// Returns a `AnyBSON` for any `BSON` value.
+    public var anyBSON: AnyBSON {
```

Comment:
I would rather we don't pollute the public API with this, as it will be on many Foundation types.